[Elixir] New concept exercise: `charlists` (#1560)
exercism/v3: languages/elixir/exercises/concept/charlists/.docs/introduction.md

Charlists are created using single quotes.

```elixir
'hello'
```

> A charlist is a list of integers where all the integers are valid code points.

Reviewer: ```suggestion
A charlist is a list of integers where all the integers are the unicode decimal value of each character: known as a code point.
```

Author: 👍, but I avoided the word "decimal" because you can as well write the code points as hexadecimal numbers.

```elixir
[65, 66, 67]
# => 'ABC'
```

You can prepend a character with `?` to get its code point.

```elixir
?A
# => 65

> 

Reviewer: ```suggestion

# any list of integers whose decimal values are comprised of printable codepoints between 1 and 255 may be displayed as a charlist, e.g.
```

Reviewer: At some point I think it may be important for students to understand why their lists of integers may be displayed as a charlist, or vice versa where their valid charlist is printed as a list of integers.

Author: I avoided mentioning that in `introduction.md` because this knowledge is not necessary to solve the exercise. I went crazy with the topic in `after.md`.

Charlists are created using single quotes.

```elixir
'hello'
```

> A charlist is a list of integers where the integers represent the Unicode values of a given character — also known as code points.

Reviewer 2: Somehow I expected the second line ("A charlist is...") to be the first line, as it tells me what charlists are. But feel free to ignore :)

Author: I totally agree that with that phrasing, the lines seem out of order. But I believe the example of a charlist using single quotes needs to go before any mention of lists of integers, so I rephrased the intro a bit to make more sense.

```elixir
[65, 66, 67]
# => 'ABC'
```

You can prepend a character with `?` to get its code point.

```elixir
?A
# => 65

[?:, ?)]
# => ':)'
```

> Because charlist are list, you can work with them just like with any other list - using recursion and pattern matching.

Reviewer 2: ```suggestion
Because charlist are lists, you can work with them just like with any other list - using recursion and pattern matching.
```

```elixir
[first_letter | _] = 'cat'
first_letter
# => 99
```

> You can concatenate two lists together using `++`.

Reviewer 2: I think this can be:

```suggestion
You can concatenate two lists using `++`.
```
